helios-services/src/test/java/com/spotify/helios/agent/SupervisorTest.java

```diff
@@ -396,12 +365,78 @@ public void verifyRuntimeExceptionSetsTaskStatusToFailed() throws Exception {
   }
 
   private Answer<?> futureAnswer(final SettableFuture<?> future) {
-    return new Answer<Object>() {
+    return (Answer<Object>) invocation -> future.get();
+  }
+
+  /**
+   * Verifies a fix for a NPE that is thrown when the Supervisor receives a goal of UNDEPLOY for a
+   * job with gracePeriod that has never been STARTed.
+   */
+  @Test
+  public void verifySupervisorHandlesUndeployingOfNotRunningContainerWithGracePeriod()
```

Comment:
and here is the actual test - the rest of the changes in SupervisorTest are refactoring to make this test possible (to use a different Job instance)